overlord/devicestate/devicemgr.go

```diff
@@ -1425,6 +1425,73 @@ func (m *DeviceManager) ensurePostFactoryReset() error {
 	return os.Remove(factoryResetMarker)
 }
 
+func (m *DeviceManager) hasUserExpired(user *auth.UserState) bool {
+	// If the user has no expiration date, then Expiration should not
+	// be set, and contain the default value.
+	if user.Expiration.IsZero() {
+		return false
+	}
+
+	return user.Expiration.Before(time.Now())
+}
+
+var osutilDelUser = osutil.DelUser
+
+func (m *DeviceManager) removeExpiredUser(st *state.State, user *auth.UserState) error {
+	// Remove the user from the system first
+	if err := osutilDelUser(user.Username, &osutil.DelUserOptions{ExtraUsers: !release.OnClassic}); err != nil {
+		return err
+	}
+
+	// And then we remove it from the auth state
+	_, err := auth.RemoveUserByUsername(st, user.Username)
+	return err
+}
+
+// ensureExpiredUsersRemoved is periodically called as a part of Ensure()
+// to remove expired users from the system.
+func (m *DeviceManager) ensureExpiredUsersRemoved() error {
```

Comment:
just `removeExpiredUsers`?

Reply:
I wanted to keep the naming convention as it was a part of ensure, but I'm not sure if that's important or if I'm just misunderstanding how we use the term, i'll happily change it if I'm using it incorrectly here

Reply:
Let's leave this to Samuele then. I suggested my preference, but it's not important :-)

Reply:
no, it's correct, all methods contributing to Ensure behavior and called from it start with ensure...